### PR TITLE
Fix issue 20638: only offer correction for module property if symbol is visible.

### PR DESCRIPTION
**Pull request by FeepingCreature**
Similar to #10867 , but easier to check.

### CODE REVIEW

File: test/fail_compilation/fail20638.d [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail20638.d(12): Error: undefined identifier `foo` in module `imports.fail20638b`
```
Comment (thewilsonator) [inline test/fail_compilation/fail20638.d]: Surely the compiler can know that it exists and is private and tell the users that.

Comment (FeepingCreature) [reply test/fail_compilation/fail20638.d]: Agreed. Separate issue though - by the time it gets to corrections, it's already too late.

Comment (thewilsonator) [reply test/fail_compilation/fail20638.d]: Really? Surely
```d
if (s && symbolIsVisible(sc, s))
{
    if (s.isPackage()) { ... }
    else { /*error with symbol hint*/ }
}
else if (s) { /*error symbol is private*/ }
else { /* error no symbol found*/ }
```

Comment (FeepingCreature) [reply test/fail_compilation/fail20638.d]: No, because `s` is the spelling-corrected symbol, not the original.

Comment (thewilsonator) [reply test/fail_compilation/fail20638.d]: Ah.

**Comment by FeepingCreature** [comment]
Note: The best solution would probably involve passing the scope down `search_correct` so that can exclude private symbols to start with. I'll take a look at that, see if it's easily done.

**Comment by FeepingCreature** [comment]
Nevermind, I misthought. When there's a direct match, you're right @thewilsonator that it's always correct to error with "tried to access private member" instead, so the question never comes up.

In the case `class Class { private void foo(); public void fob(); }`, `obj.foo` should always print a "private member not accessible" error, not go searching for misspellings, so the fact that the misspelling doesn't point it at fob is not an issue.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**